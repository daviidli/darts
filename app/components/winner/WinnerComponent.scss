.container {
	height: 100vh;
	width: 100vw;
	margin: 0;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	.message {
		font-size: 4em;
		text-align: center;
	}

	.name {
		font-size: 8em;
		font-weight: 900;
		text-shadow: 0.06em 0.06em #e49708;
		text-align: center;
	}

	.button {
		margin-top: 5em;
	}
}
