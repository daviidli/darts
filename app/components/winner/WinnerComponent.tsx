import React from 'react';
import { useHistory } from 'react-router-dom';
import { Typography, Button } from '@material-ui/core';
import { playerType } from '../../reducers/types';
import styles from './WinnerComponent.scss';
import routes from '../../constants/routes.json';

export type Props = {
	winner: number;
	players: playerType[];
	setWinner: (index: number) => {};
};

const WinnerComponent = (props: Props) => {
	const { winner, players, setWinner } = props;
	const history = useHistory();

	const handleClick = () => {
		history.push(routes.HOME);
		setWinner(-1);
	};

	return (
		<div className={styles.container}>
			<Typography className={styles.message}>Winner is</Typography>
			<Typography className={styles.name}>
				{players[winner].name}
			</Typography>
			<Button
				variant="contained"
				color="secondary"
				className={styles.button}
				onClick={handleClick}
			>
				Home
			</Button>
		</div>
	);
};

export default WinnerComponent;
