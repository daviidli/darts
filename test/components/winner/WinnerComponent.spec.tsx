import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import WinnerComponent, {
	Props
} from '../../../app/components/winner/WinnerComponent';

Enzyme.configure({ adapter: new Adapter() });

const mockPush = spy();
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useHistory: () => ({
		push: mockPush
	})
}));

const initialProps: Props = {
	winner: 0,
	players: [{ id: '1', name: 'player1' }],
	setWinner: () => ({})
};

const setup = (props: Props = initialProps) => {
	const wrapper = shallow(<WinnerComponent {...props} />);
	return {
		wrapper,
		player: wrapper.find('.name'),
		button: wrapper.find('.button')
	};
};

describe('Winner component', () => {
	it('should match the snapshot', () => {
		const tree = renderer
			.create(<WinnerComponent {...initialProps} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should show correct player name', () => {
		const setWinnerMock = spy();
		const props: Props = {
			winner: 1,
			players: [
				{
					id: '1',
					name: 'player1'
				},
				{
					id: '2',
					name: 'player2'
				}
			],
			setWinner: setWinnerMock
		};
		const { player } = setup(props);
		expect(player.at(0).text()).toMatch('player2');
		expect(setWinnerMock.notCalled).toBeTruthy();
	});

	it('should go to settings screen on click', () => {
		const setWinnerMock = spy();
		const props = { ...initialProps };
		props.setWinner = setWinnerMock;
		const { button } = setup(props);
		expect(button.at(0).text()).toMatch('Home');
		button.at(0).simulate('click');
		expect(mockPush.callCount).toBe(1);
		expect(mockPush.calledWith('/')).toBeTruthy();
		expect(setWinnerMock.calledOnce).toBeTruthy();
		expect(setWinnerMock.calledWith(-1)).toBeTruthy();
	});
});
