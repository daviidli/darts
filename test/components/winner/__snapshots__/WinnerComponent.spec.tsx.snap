// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner component should match snapshot 1`] = `
<div
  className="container"
>
  <WithStyles(ForwardRef(Typography))
    className="message"
  >
    Winner is
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    className="name"
  >
    player1
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Button))
    className="button"
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    Home
  </WithStyles(ForwardRef(Button))>
</div>
`;
